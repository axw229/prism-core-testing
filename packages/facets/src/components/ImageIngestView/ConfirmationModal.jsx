// @flow
import React, { useRef, useCallback, useEffect } from 'react'
import { Link, useHistory } from 'react-router-dom'
import { FormattedMessage } from 'react-intl'

const baseClass = 'confirmation-modal'
const wrapperClass = `${baseClass}__wrapper`
const containerClass = `${baseClass}__container`
const contentClass = `${baseClass}__content`
const buttonClass = `${baseClass}__button`

type Props = {
  onClickNo: Function
}

const ConfirmationModal = ({ onClickNo }: Props) => {
  const btnYesRef = useRef()
  const btnNoRef = useRef()
  const history = useHistory()

  useEffect(() => {
    if (btnYesRef && btnYesRef.current) btnYesRef.current.focus()
  }, [])

  const blurHandler = useCallback((ref: any) => {
    if (ref.current === btnYesRef.current) {
      btnNoRef.current.focus()
    } else if (ref.current === btnNoRef.current) {
      btnYesRef.current.focus()
    }
  }, [btnYesRef, btnNoRef])

  return (
    <div className={`${wrapperClass}`}>
      <div className={`${containerClass}`}>
        <p className={`${contentClass}`}>
          <FormattedMessage id='CONFIRMATION_DIALOG_MATCH_A_PHOTO_EXIT' />
        </p>
        <Link tabIndex='-1' to={'/active'}>
          <button ref={btnYesRef} onBlur={() => blurHandler(btnYesRef)} className={`${buttonClass}`} onClick={() => history.push('/active')}>
            <FormattedMessage id='YES' />
          </button>
        </Link>
        <button ref={btnNoRef} onBlur={() => blurHandler(btnNoRef)} className={`${buttonClass}`} onClick={onClickNo}><FormattedMessage id='NO' /></button>
      </div>
    </div>
  )
}

export {
  contentClass,
  buttonClass
}
export default ConfirmationModal
