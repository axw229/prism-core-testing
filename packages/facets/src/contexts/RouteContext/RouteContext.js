import React from 'react'
export const RouteContext = React.createContext('defaultValue')
export const RouteConsumer = RouteContext.Consumer
