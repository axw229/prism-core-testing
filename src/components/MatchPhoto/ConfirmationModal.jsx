// @flow
import React, { useContext, useRef, useCallback, useEffect } from 'react'
import { Link } from 'react-router-dom'
import { RouteContext } from '../../contexts/RouteContext/RouteContext'
import { FormattedMessage } from 'react-intl'

const baseClass = 'confirmation-modal'
const wrapperClass = `${baseClass}__wrapper`
const containerClass = `${baseClass}__container`
const contentClass = `${baseClass}__content`
const buttonClass = `${baseClass}__button`

type Props = {
  onClickNo: Function
}

const ConfirmationModal = ({ onClickNo }: Props) => {
  const routeContext = useContext(RouteContext)
  const btnYesRef = useRef()
  const btnNoRef = useRef()

  useEffect(() => {
    if (btnYesRef && btnYesRef.current) btnYesRef.current.focus()
  }, [])

  const blurHandler = useCallback((ref: any) => {
    if (ref.current === btnYesRef.current) {
      btnNoRef.current.focus()
    } else if (ref.current === btnNoRef.current) {
      btnYesRef.current.focus()
    }
  }, [btnYesRef, btnNoRef])

  return (
    <div className={`${wrapperClass}`}>
      <div className={`${containerClass}`}>
        <p className={`${contentClass}`}>
          <FormattedMessage id='CONFIRMATION_DIALOG_MATCH_A_PHOTO_EXIT' />
        </p>
        <Link tabIndex='-1' to={`/active`}>
          <button ref={btnYesRef} onBlur={() => blurHandler(btnYesRef)} className={`${buttonClass}`} onClick={() => routeContext.setActiveComponent()}><FormattedMessage id='YES' /></button>
        </Link>
        <button ref={btnNoRef} onBlur={() => blurHandler(btnNoRef)} className={`${buttonClass}`} onClick={onClickNo}><FormattedMessage id='NO' /></button>
      </div>
    </div>
  )
}

export {
  contentClass,
  buttonClass
}
export default ConfirmationModal
