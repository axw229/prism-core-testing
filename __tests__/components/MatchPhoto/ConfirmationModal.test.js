import React from 'react'
import ConfirmationModal from 'src/components/MatchPhoto/ConfirmationModal'
import { fireEvent } from '@testing-library/dom'
import { RouteContext } from 'src/contexts/RouteContext/RouteContext'

test('ConfirmationModal', async () => {
  const onClickNo = jest.fn()
  const setActiveComponent = jest.fn()
  const { getByText } = render(
    <RouteContext.Provider value={{ setActiveComponent: setActiveComponent }}>
      <ConfirmationModal
        onClickNo={onClickNo}
      />
    </RouteContext.Provider>
  )

  getByText('The photo content will be lost if you close. Make sure the colors you want to keep have been added to your palette. Do you still want to close?')
  getByText('Yes')
  getByText('No')
  await fireEvent.click(getByText('Yes'))
  expect(setActiveComponent).toHaveBeenCalled()
})
