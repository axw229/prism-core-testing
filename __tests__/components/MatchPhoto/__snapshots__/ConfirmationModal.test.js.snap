// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModal with props should match snapshot with props 1`] = `
<div
  className="confirmation-modal__wrapper confirmation-modal__wrapper--active"
>
  <div
    className="confirmation-modal__container"
  >
    <p
      className="confirmation-modal__content"
    >
      The photo content will be lost if you close. Make sure the colors you want to keep have been added to your palette. Do you still want to close?
    </p>
    <Link
      to="/active"
    >
      <button
        className="confirmation-modal__button"
      >
        YES
      </button>
    </Link>
    <button
      className="confirmation-modal__button"
      onClick={[MockFunction]}
    >
      NO
    </button>
  </div>
</div>
`;
